chromeStorageApi, storageService unit tests

m0t0r opened the pull request:
I have fixed all issues and took into account comments which were presented in the first pull request. 
This pull request contains:
- chromeStorageApi wrapper with unit tests
- storageService refactoring and unit tests

m0t0r committed:
chromeStorageApi, storageService unit tests

jjal commented on app/index.html:
```diff
@@ -74,8 +74,8 @@
-<script src="scripts/services/chromeStorageApi.js"></script>
-<script src="scripts/services/storageService.js"></script>
+<script src="scripts/services/chrome-storage-api.js"></script>
+<script src="scripts/services/storage-service.js"></script>
```
:+1:

jjal commented:
- please rebase and re-issue
- new tests look good but fail locally for me - dependencies
- please update bower / karma conf with your sinon stuff

Commit:
Merge pull request #147 from jofomah/refactor/redesign-counter-ui

Refactor/redesign counter ui

m0t0r committed:
merge

m0t0r committed:
unit test fixes

m0t0r committed:
merge

m0t0r commented:
I have fixed issues. All unit tests have been passed. I think you can finally merge with develop. :)

Also, I removed siononJS from my test. So, seems we don't need it, jasmine spies work fine.

tlvince commented:
To avoid superfluous merge commits, next time, you could use `git pull --rebase origin/develop`.

tlvince commented on test/mock/chrome-storage-api.js:
```diff
@@ -0,0 +1,56 @@
+  beforeEach(module('lmisChromeApp'));
+
+  beforeEach(function () {
+    mockWindow = {
```
With regards to my previous comment about moving the spec in to the `/mock` directory, I meant just the `mockwindow` object, not the spec itself. This should be moved back into the `/spec/services` directory. Sorry for the confusion.

m0t0r replied:
Alright. I will fix it

jjal commented:
When I merge this locally I get 

```
TypeError: Object.keys called on non-object
 at Function.keys (native)
at chrome-extension://.../scripts/services/storage-service.js:145:30
```

which is in getAll.- there is a test for that function but for some reason it doesn't catch this.
It's blocking saves of stock count so we'll need to figure that out before merging it in.

m0t0r committed:
chromeStorageApi: get method fix
